pygbm/grower.py

```diff
@@ -336,6 +339,7 @@ def split_next(self):
         node = heappop(self.splittable_nodes)
 
         tic = time()
+        split_indices = split_indices_parallel if self.parallel_splitting else split_indices_single_thread
```

Comment:
I think we should fork in `SplittingContext` instead of in the grower. We probably don't need `grower.parallel_splitting`.

That is, call `SplittingContext.split_indice()` in the grower, and in splitting.py do:


```python
def split_indice(...)
   if self.parallel_splitting:
       return ...
   else:
       return ...
````